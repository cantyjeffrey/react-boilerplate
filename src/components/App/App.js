/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div>
        <h1>Application</h1>
      </div>
    );
  }
}

export default App;
