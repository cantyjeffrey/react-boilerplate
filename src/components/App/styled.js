// @flow

import styled from 'styled-components';

export const Root = styled.div``;

export default {};
