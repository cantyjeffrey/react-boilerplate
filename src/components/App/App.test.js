// @flow
import { shallow } from 'enzyme';
import 'jest-styled-components';
import puppeteer from 'puppeteer';
import React from 'react';
import App from './App';

const PROTOCOL = process.env.HTTPS === 'true' ? 'https' : 'http';
const PORT = parseInt(process.env.PORT, 10) || 3000;
const HOST = process.env.HOST || '0.0.0.0';
const URL = `${PROTOCOL}://${HOST}:${PORT}`;

describe('<App />', () => {
  let browser;

  beforeAll(async () => {
    browser = await puppeteer.launch();
  });

  it('should match existing snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match existing image snapshot', async () => {
    const page = await browser.newPage();
    await page.setViewport({ width: 1366, height: 768 });
    await page.goto(`${URL}`);
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      customSnapshotIdentifier: 'App',
    });
  });

  it('renders a headline with correct copy', () => {
    const wrapper = shallow(<App />);
    const headline = <h1>Application</h1>;
    expect(wrapper).toContainReact(headline);
  });

  it('renders with correct styles', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toHaveStyleRule('border', '1px solid red');
  });

  afterAll(async () => {
    await browser.close();
  });
});
