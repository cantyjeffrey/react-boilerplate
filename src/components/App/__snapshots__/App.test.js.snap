// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match existing snapshot 1`] = `
<styled.div>
  <h1>
    Application
  </h1>
</styled.div>
`;
